File: day_7/10_speckled_frogs.js
frogs = Array(1..10).reverse

frogs.each do |x|
  if x > 1
    puts "#{x} speckled frogs sat on a log eating some most delicious bugs. One jumped in the pool where its nice and cool, then there were #{x - 1} speckled frogs."
 else
    puts "#{x} speckled frog sat on a log eating some most delicious bugs.One jumped in the pool where its nice and cool,then there were no more speckled frogs!"
 end
end


# Extension 1
#Print word versions of each number in the first and fourth lines, for example, the first verse in the above example would print 'Three speckled frogs...' and 'were two speckled frogs'.

frog_num = {
  10 => "ten",
  9 => "nine",
  8 => "eight",
  7 => "seven",
  6 => "six",
  5 => "five",
  4 => "four",
  3 => "three",
  2 => "two",
  1 => "one"
}

frog_num.each { |key, value|
  if key > 1
    puts "#{value.capitalize} speckled frogs sat on a log eating some most delicious bugs. One jumped in the pool where its nice and cool, then there were " + frog_num[key-1] + " speckled frogs. \n"
  else
    puts "#{value.capitalize} speckled frog sat on a log eating some most delicious bugs.One jumped in the pool where its nice and cool,then there were no more speckled frogs!"
  end
}

# Extension 2
#Make your program work for any number of frogs.

puts "How many frogs are on the log? (e.g. 10)"
user_frog_num = gets.chomp.to_i

user_frog_num2 = (1..user_frog_num).reverse_each

user_frog_num2.each do |y|
  if y > 1
    puts "#{y} speckled frogs sat on a log eating some most delicious bugs. One jumped in the pool where its nice and cool, then there were #{y - 1} speckled frogs."
 else
    puts "#{y} speckled frog sat on a log eating some most delicious bugs.One jumped in the pool where its nice and cool,then there were no more speckled frogs!"
 end
end

File: day_7/day_7_files/10_speckled_frogs.rb
# require "humanize"

def speckled_frogs(frog_num)
  num_to_word = {0 => "zero", 1 => "one", 2 => "two", 3 => "three", 4 => "four", 5 => "five", 6 => "six", 7 => "seven", 8 => "eight", 9 => "nine", 10 => "ten" }
  # num_to_word = Hash{1..100((1..100).humanize)}
  if frog_num > 2
    puts "#{num_to_word.fetch(frog_num).capitalize} speckled frogs sat on a log eating some most delicious bugs. One jumped in the pool where its nice and cool, then there were #{num_to_word.fetch(frog_num - 1).downcase} speckled frogs."
  elsif frog_num == 2
    puts "#{num_to_word.fetch(frog_num).capitalize} speckled frogs sat on a log eating some most delicious bugs. One jumped in the pool where its nice and cool, then there was #{num_to_word.fetch(frog_num - 1).downcase} speckled frog."
  else frog_num == 1
    puts "#{num_to_word.fetch(frog_num).capitalize} speckled frog sat on a log eating some most delicious bugs. It jumped in the pool where its nice and cool, then there were #{num_to_word.fetch(frog_num - 1).downcase} speckled frogs!"
  end
end

print "How many speckled frogs are hanging out on the log?: "
max_frogs = gets.chomp.to_i
speckled_frogs(max_frogs)
